### PR TITLE
Frontend refactor

### PR DESCRIPTION
**Pull request by zurdi15**
General front code refactor + UI rework + improved UX:

- Redesigned drawer menu sections
- Added general action buttons for database search and games upload
- Changed gallery filters to a new secondary drawer 
- Added new drawer for firmware/bios management
- Reworked gallery table view and users managing view
- Reworked dialogs
- Reworked notes section
- Reworked saves/states section
- Reworked emulatorjs player view:
  - Added back to gallery button
  - Added back to details button
- Added 404 views when not finding/matching platforms or games

### CODE REVIEW

File: backend/endpoints/firmware.py [hunk 1]
```diff
@@ -14,7 +14,9 @@
 
 @protected_route(router.post, "/firmware", ["firmware.write"])
 def add_firmware(
-    request: Request, platform_id: int, files: list[UploadFile] | None = None
+    request: Request,
+    platform_id: int,
+    files: list[UploadFile] = File(...),  # noqa: B008
```
Comment (gantoine) [inline backend/endpoints/firmware.py:19]: 👍🏼

File: frontend/src/components/Dialog/Asset/DeleteAssets.vue [hunk 2]
```diff
@@ -1,217 +1,243 @@
...
+    key: "file_name",
+  },
+] as const;
+const page = ref(1);
+const itemsPerPage = ref(10);
+const pageCount = ref(0);
+const PER_PAGE_OPTIONS = [10, 25, 50, 100];
 
+// Funtcions
+// TODO: make remove assets reactive
```
Comment (gantoine) [inline frontend/src/components/Dialog/Asset/DeleteAssets.vue:51]: one day we should talk about reactivity for platforms and galleries

File: frontend/src/components/Dialog/Asset/DeleteAssets.vue [hunk 3]
```diff
@@ -1,217 +1,243 @@
...
+            (asset) => !deletedAssetIds.includes(asset.id)
+          ) ?? [];
         romsStore.update(romRef.value);
         emitter?.emit("romUpdated", romRef.value);
       }
     })
     .catch(({ response, message }) => {
       emitter?.emit("snackbarShow", {
-        msg: `Unable to delete saves: ${
+        msg: `Unable to delete ${assetType.value.slice(5)}: ${
```
Comment (gantoine) [inline frontend/src/components/Dialog/Asset/DeleteAssets.vue]: i dont love this, personally i'd just do `assetType.value` even if it looks weird to users. or maybe map it like `{user_saves: "saves"}`

File: frontend/src/components/Dialog/Platform/DeleteFirmware.vue [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    v-model="show"
+    icon="mdi-delete"
+    scroll-content
+    :width="mdAndUp ? '60vw' : '95vw'"
+  >
+    <template #prepend>
+      <v-list-item class="text-center">
+        <span>Removing the following</span>
+        <span class="text-romm-accent-1 mx-1">{{ firmwares.length }}</span>
+        <span>firmwares from RomM. Do you confirm?</span>
```
Comment (gantoine) [inline frontend/src/components/Dialog/Platform/DeleteFirmware.vue]: use either "firmware", "bios files" or "firmware files". also i'd rephrase this as "Confirm the removal of the following X firmware files from RomM"

File: frontend/src/components/Dialog/Rom/SearchRom.vue [hunk 5]
```diff
@@ -1,88 +1,95 @@
...
+    // Auto hide android keyboard
+    const inputElement = document.getElementById("search-text-field");
+    inputElement?.blur();
+    searching.value = true;
+    searched.value = true;
+    searchedRoms.value = (
+      await romApi.getRoms({ searchTerm: searchValue.value })
+    ).data.sort((a, b) => {
+      return a.platform_name.localeCompare(b.platform_name);
+    });
```
Comment (gantoine) [inline frontend/src/components/Dialog/Rom/SearchRom.vue:68]: sorting by platform name as opposed to game name?

File: frontend/src/components/Dialog/User/CreateUser.vue [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
-
 import userApi from "@/services/api/user";
 import storeUsers from "@/stores/users";
+import type { Events } from "@/types/emitter";
+import type { Emitter } from "mitt";
+import { inject, ref } from "vue";
+import { useDisplay } from "vuetify";
+import RDialog from "@/components/common/Dialog.vue";
 
+// Props
```
Comment (gantoine) [inline frontend/src/components/Dialog/User/CreateUser.vue:10]: are these comments really necessary?

File: frontend/src/components/Drawer/Footer.vue [hunk 7]
```diff
@@ -100,37 +95,38 @@ async function logout() {
   <v-list-item
     v-if="
       GITHUB_VERSION &&
-        VERSION !== GITHUB_VERSION &&
-        !latestVersionDismissed &&
-        !rail
+      VERSION < GITHUB_VERSION &&
```
Comment (gantoine) [inline frontend/src/components/Drawer/Footer.vue:98]: will this work with semantic versions? you might need to use a semantic version lib to compare them

https://stackoverflow.com/a/30256597/1179430

File: frontend/src/components/Gallery/AppBar/FilterTextField.vue [hunk 8]
```diff
@@ -3,34 +3,33 @@ import storeGalleryFilter from "@/stores/galleryFilter";
...
 // Functions
 const filterRoms = debounce(() => {
-  galleryFilter.setFilterSearch(filter?.value ?? "");
-  emitter?.emit('filter', null);
+  emitter?.emit("filter", null);
 }, 500);
 
-onMounted(() => {
-  filter.value = galleryFilter.filterSearch;
-});
```
Comment (gantoine) [inline frontend/src/components/Gallery/AppBar/FilterTextField.vue:23]: no longer needed to clean on mount?

File: frontend/src/components/Gallery/AppBar/ScanBtn.vue [hunk 9]
```diff
@@ -15,24 +15,17 @@ async function scan() {
   if (!socket.connected) socket.connect();
 
   socket.emit("scan", {
-    platforms: [romsStore.platformID],
+    platforms: [romsStore.currentPlatform?.id],
```
Comment (gantoine) [inline frontend/src/components/Gallery/AppBar/ScanBtn.vue:18]: 👍🏼

File: frontend/src/views/Play/Player.vue [hunk 10]
```diff
@@ -17,6 +19,7 @@ const props = defineProps<{
 const romRef = ref<DetailedRom>(props.rom);
 const saveRef = ref<SaveSchema | null>(props.save);
 const stateRef = ref<StateSchema | null>(props.state);
+const emitter = inject<Emitter<Events>>("emitter");
```
Comment (gantoine) [inline frontend/src/views/Play/Player.vue]: is this used?

File: frontend/src/views/GameDetails.vue [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+onBeforeMount(async () => {
+  emitter?.emit("showLoadingDialog", { loading: true, scrim: false });
+  if (rom.value) {
+    emitter?.emit("showLoadingDialog", { loading: false, scrim: false });
+  } else {
+    await fetchDetails();
+  }
+  const downloadStore = storeDownload();
+  downloadStore.clear();
```
Comment (gantoine) [inline frontend/src/views/GameDetails.vue:81]: this download store clearing, was it copied from the old file?

File: frontend/src/stores/roms.ts [hunk 12]
```diff
@@ -15,48 +16,44 @@ export type DetailedRom = DetailedRomSchema;
 
 export default defineStore("roms", {
   state: () => ({
-    _platformID: 0,
-    _all: [] as SimpleRom[],
+    currentPlatform: null as Platform | null,
```
Comment (gantoine) [inline frontend/src/stores/roms.ts:19]: there's a reason i used platformID internally, instead of the platform... if i remember ill update this comment

File: frontend/src/components/common/Notification.vue [hunk 13]
```diff
@@ -1,29 +1,39 @@
...
 }
 </script>
 
 <template>
   <v-snackbar
+    transition="scroll-y-transition"
     v-model="show"
-    :timeout="snackbarStatus.timeout ? snackbarStatus.timeout : 2000"
-    location="top"
+    :timeout="snackbarStatus.timeout ? snackbarStatus.timeout : 3000"
```
Comment (gantoine) [inline frontend/src/components/common/Notification.vue]:
```suggestion
    :timeout="snackbarStatus.timeout || 3000"
```

File: frontend/src/components/Game/Card/Base.vue [hunk 14]
```diff
@@ -1,60 +1,212 @@
...
+import storeGalleryView from "@/stores/galleryView";
+import storeRoms, { type SimpleRom } from "@/stores/roms.js";
+import { storeToRefs } from "pinia";
+import { onMounted, ref } from "vue";
+import { useTheme } from "vuetify";
+
+// Type guard function to check if rom is of type SimpleRom or SearchRomSchema
+function isSimpleRom(rom: SimpleRom | SearchRomSchema): rom is SimpleRom {
+  return (rom as SimpleRom).id !== undefined;
+}
```
Comment (gantoine) [inline frontend/src/components/Game/Card/Base.vue:15]: 👍🏼

**Review by gantoine — CHANGES_REQUESTED** [review]
-3K lines, very nice 👏🏼 

just a couple small comments
i didn't review the jsx very closely since much of it was copied over, but i trust it works

File: frontend/src/views/Play/Player.vue (hunk 10 above)
Comment (zurdi15) [reply frontend/src/views/Play/Player.vue]: I'll check, probably we have now some variables not being used anymore due to the refactor

File: frontend/src/components/Dialog/User/CreateUser.vue (hunk 6 above)
Comment (zurdi15) [reply frontend/src/components/Dialog/User/CreateUser.vue:10]: Those helps me a lot when working on front :/

File: frontend/src/components/Dialog/Rom/SearchRom.vue (hunk 5 above)
Comment (zurdi15) [reply frontend/src/components/Dialog/Rom/SearchRom.vue:68]: Yeah I kind agree with this, sorting by platform and then by name?

File: frontend/src/components/Drawer/Footer.vue (hunk 7 above)
Comment (zurdi15) [reply frontend/src/components/Drawer/Footer.vue:98]: Yeah I tested it and it works pretty well!

File: frontend/src/components/Gallery/AppBar/FilterTextField.vue (hunk 8 above)
Comment (zurdi15) [reply frontend/src/components/Gallery/AppBar/FilterTextField.vue:23]: Not anymore, now all the filters are cleaned from the gallery

File: frontend/src/views/GameDetails.vue (hunk 11 above)
Comment (zurdi15) [reply frontend/src/views/GameDetails.vue:81]: hmm yeah not sure about this one, i'll check!

File: frontend/src/stores/roms.ts (hunk 12 above)
Comment (zurdi15) [reply frontend/src/stores/roms.ts:19]: i tried every possible way to navigate through romm (i think) and coudn't find any issue using the platform

File: frontend/src/components/Dialog/Rom/SearchRom.vue (hunk 5 above)
Comment (zurdi15) [reply frontend/src/components/Dialog/Rom/SearchRom.vue:68]: Okay i have been testing to be sure and this already orders by platform and then by game name!

File: frontend/src/views/GameDetails.vue (hunk 11 above)
Comment (zurdi15) [reply frontend/src/views/GameDetails.vue:81]: okay it's needed to clear the multipart selector

File: frontend/src/stores/roms.ts (hunk 12 above)
Comment (zurdi15) [reply frontend/src/stores/roms.ts:19]: And of course checked every place where we were using _platformID and no issues at all

**Review by gantoine — APPROVED** [review]
